rewrite-maven/src/main/java/org/openrewrite/maven/tree/ResolvedPom.java

```diff
@@ -89,6 +94,16 @@ public ResolvedPom(Pom requested, Iterable<String> activeProfiles) {
         this.subprojects = subprojects;
     }
 
+    private static List<ResolvedManagedDependency> sortDependencyManagement(List<ResolvedManagedDependency> sorted) {
+        if (sorted.isEmpty()) {
+            return sorted;
+        }
+
+        // TimSort is O(n) for already sorted data
+        sorted.sort(MANAGED_DEPENDENCY_COMPARATOR);
+        return sorted;
+    }
```

Comment:
Hmm I'm not sure if it is correct / matches Maven's behavior to sort dependency management.
Maven's version conflict resolution strategy is "nearest wins" but if there are two equally-near options it _probably_ comes down to order, although I don't know off the top of my head if it would be first wins or last wins.

If something other than order determines which version of jackson-core wins in a situation like this one then it's probably fine
```
org:transitive-jackson-dep
+- org:depends-on-jackson
|  +- com.fasterxml.jackson.core:jackson-core:2.9.9:compile
+- org:also-depends-on-jackson
   +- com.fasterxml.jackson.core:jackson-core:2.12.0:compile
```

Reply:
I think here whichever dependency comes first (here 2.9.9) should win. I don't think the sorting should affect this, as the sorting isn't performed until after the resolution. But it would make sense to add a test for this.